mondrian/src/main/java/mondrian/olap/fun/CrossJoinFunDef.java

```diff
@@ -999,29 +999,27 @@ private static boolean checkData(
         Evaluator evaluator)
     {
         if (cnt < 0) {
-            // no measures found, use standard algorithm
-            if (measureSet.isEmpty()) {
-                Object value = evaluator.evaluateCurrent();
-                if (value != null
-                    && !(value instanceof Throwable))
-                {
-                    return true;
-                }
-            } else {
+            boolean found = false;
+            Object value = evaluator.evaluateCurrent();
+            if (value != null
+                && !(value instanceof Throwable))
+            {
+                found = true;
+            }
+            if (!measureSet.isEmpty()) {
```

Comment:
@mkambol Yep, I can explain: the problem was that we set a measures from measureSet (which haven't any calculated measures in it) in context and then evaluate evaluator on it. Because of this we haven't received any "true" statement that says us that "yep, we found a non-null value of some combination of Members from nonAllMembers and Measures from measureSet". But if we use "_standard algorithm_" first (without setting non-calculated measures to the context) we will receive "true". (Which was omitted before my fix)